src/monitoring_tests/combinatorial_auction_surplus_test.py

```diff
@@ -0,0 +1,291 @@
+        correspond to those baselines that would have resulted in filtering of the solution. This
+        information is used e.g. to trigger alerts depending on which solver resulted in filtering.
+        """
+        result: list[list[int]] = []
+        for aggregate_solution in aggregate_solutions:
+            flag = []
+            for token_pair, surplus in aggregate_solution.items():
+                if token_pair in baseline_surplus:
+                    surplus_ref, solution_index = baseline_surplus[token_pair]
+                    if surplus < surplus_ref:
+                        flag.append(solution_index)
+            result.append(flag)
+
+        return result
+
+    def determine_winning_solutions(
+        self,
+        aggregate_solutions: list[dict[tuple[str, str], Fraction]],
+        baseline_surplus: dict[tuple[str, str], tuple[Fraction, int]],
+        filter_mask: list[list[int]],
+    ) -> dict[tuple[str, str], int]:
+        """Determine the winning solutions for the different token pairs.
+        There is one batch winner, and the remaining token pairs are won by single aggregate order
+        solutions.
+        """
+        result: dict[tuple[str, str], int] = {}
+        # one batch winner
+        for i, _ in reversed(list(enumerate(aggregate_solutions))):
+            if len(filter_mask[i]) == 0:
+                batch_winner_index = i
+                break
+        for token_pair, _ in aggregate_solutions[batch_winner_index].items():
+            result[token_pair] = batch_winner_index
+        # the remaining token pairs are won by baseline solutions
+        for token_pair, surplus_and_index in baseline_surplus.items():
+            if not token_pair in result:
+                result[token_pair] = surplus_and_index[1]
+        return result
+
+    def determine_winning_solvers(
+        self,
+        solutions: list[dict[str, Any]],
+        aggregate_solutions: list[dict[tuple[str, str], Fraction]],
+        winning_solutions: dict[tuple[str, str], int],
+    ) -> dict[str, dict[tuple[str, str], Fraction]]:
+        """Determine the winning solvers and the surplus for each token pair.
+        There is one batch winner, and the remaining token pairs are won by single aggregate order
+        solutions.
+        """
+        result: dict[str, dict[tuple[str, str], Fraction]] = {}
+        for token_pair, solution_index in winning_solutions.items():
+            solution = solutions[solution_index]
+            solver = solution["solver"]
+            if not solver in result:
+                result[solver] = {}
+            result[solver][token_pair] = aggregate_solutions[solution_index][token_pair]
+        return result
+
+    def run(self, tx_hash: str) -> bool:
+        """
+        Wrapper function for the whole test. Checks if solver competition data is retrievable
+        and runs EBBO test, else returns True to add to list of unchecked hashes.
```

Comment:
>> Runs EBBO test..

Maybe fix this.

>> else returns True...
is this a typo or is this the convention here?

Reply:
I fixed it. The same error is present in some other test. Will fix that later.